homeassistant/components/ozw/climate.py

```diff
@@ -0,0 +1,388 @@
+        return [PRESET_NONE] + [
+            val[VALUE_LABEL]
+            for val in self.values.mode.value[VALUE_LIST]
+            if val[VALUE_ID] > 10
+        ]
+
+    @property
+    def swing_mode(self):
+        """Return the swing setting."""
+        if not self.values.swing_mode:
+            return None
+        return self.values.swing_mode.value
+
+    @property
+    def swing_modes(self):
+        """Return the list of available swing modes."""
+        if not self.values.swing_mode:
+            return []
+        return [
+            entry[VALUE_LABEL] for entry in self.values.swing_mode.value[VALUE_LIST]
+        ]
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        if len(self._current_mode_setpoint_values) != 1:
+            return None
+        return self._current_mode_setpoint_values[0].value
+
+    @property
+    def target_temperature_low(self) -> Optional[float]:
+        """Return the lowbound target temperature we try to reach."""
+        if len(self._current_mode_setpoint_values) < 2:
+            return None
+        return self._current_mode_setpoint_values[0].value
+
+    @property
+    def target_temperature_high(self) -> Optional[float]:
+        """Return the highbound target temperature we try to reach."""
+        if len(self._current_mode_setpoint_values) < 2:
+            return None
+        return self._current_mode_setpoint_values[1].value
+
+    async def async_set_temperature(self, **kwargs):
+        """Set new target temperature.
+
+        Must know if single or double setpoint.
+        """
+        if len(self._current_mode_setpoint_values) == 1:
+            setpoint = self._current_mode_setpoint_values[0]
+            target_temp = kwargs.get(ATTR_TEMPERATURE)
+            if setpoint is not None and target_temp is not None:
+                setpoint.send_value(target_temp)
+        elif len(self._current_mode_setpoint_values) == 2:
+            (setpoint_low, setpoint_high) = self._current_mode_setpoint_values
+            target_temp_low = kwargs.get(ATTR_TARGET_TEMP_LOW)
+            target_temp_high = kwargs.get(ATTR_TARGET_TEMP_HIGH)
+            if setpoint_low is not None and target_temp_low is not None:
+                setpoint_low.send_value(target_temp_low)
+            if setpoint_high is not None and target_temp_high is not None:
+                setpoint_high.send_value(target_temp_high)
+
+    async def async_set_fan_mode(self, fan_mode):
+        """Set new target fan mode."""
+        if not self.values.fan_mode:
+            return
+        # get id for this fan_mode
+        fan_mode_value = _get_list_id(self.values.fan_mode.value[VALUE_LIST], fan_mode)
+        if fan_mode_value is None:
+            _LOGGER.warning("Received an invalid fan mode: %s", fan_mode)
+            return
+        self.values.fan_mode.send_value(fan_mode_value)
+
+    async def async_set_swing_mode(self, swing_mode):
```

Comment:
We need to test this too. Do we have fixture data?

Reply:
No, I don't have swing mode CC in any of the dumps that are provided. I've removed the code for now. It can be added back on user demand.